### PR TITLE
Add Node "editor-only" Toggle

### LINKED ISSUE #7715: Add a way to disable/deactivate a node in the editor
### Describe the project you are working on

When developing a game sometimes I want to set a node and it's children to an inactive state so I can test better without having to delete the entire node and losing track of what I removed.

### Describe the problem or limitation you are having in your project

I cannot disable/deactivate nodes in the editor right now. This causes me to lose track of which nodes I deleted when I am testing something and I want a node to be gone for testing.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

If I can disable/deactivate a node I can visually still see the node in the scene view. Then I have a clear view of which nodes are active and which are not.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

In the inspector view there will be a disable/deactivate checkbox. If this is checked the node becomes faded or another color in the scene view.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I think it will be used quite often.

### Is there a reason why this should be core and not an add-on in the asset library?

I think it is quite an essential part of being able to test.

### PR DESCRIPTION
**Pull request by torcado194**
Closes https://github.com/godotengine/godot-proposals/issues/7715
Related: https://github.com/godotengine/godot-proposals/issues/3433

This PR adds editor functionality to toggle "editor-only" mode for Nodes in the scene tree, allowing you to omit (or "comment out") nodes when running or exporting the project. This is especially useful for prototyping, as it allows users to quickly test and compare different nodes/scenes.

This is implemented via editor metadata, which the scene packer checks for to replace relevant nodes with empty Node objects when building. This prevents node path references from erroring.

The editor functionality is contained in a context menu option, which is assignable to a hotkey for quick access.

demo2

<details><summary>Original description</summary>
<p>

This PR adds an additional toggle to nodes in the tree, allowing you to toggle the existence of the node when running or exporting the project. This allows for **editor-only** nodes and **completely removed** nodes, when combined with the existing visibility toggle.

This is one of the first changes I made to Godot locally, about a year ago. I find it invaluable for prototyping. I've since given the code to friends and anyone who's asked for it. I figured enough people have wanted it that I should submit a PR for it.

preview

(This could alternatively be hidden behind the right-click menu. I opted for this solution since I primarily use it for quick prototyping, so having immediate access is preferred.)

Note about the implementation:
I originally implemented this as an additional `tag` on the node resource, so that when parsing the node in `resource_format_text.cpp` it could apply the `disabled` state directly so that `SceneState::instantiate()` could read that data up front rather than needing to parse the properties. The metadata implementation was used instead because I did not want to interfere with the binary saving/loading system, which is necessary for keeping the behavior when exporting. I still believe the tag is the better solution, if anyone has insights about this I would appreciate it.

</p>
</details>

### CODE REVIEW

**Comment by AdriaandeJongh** [comment]
In terms of UX, this is another icon in a place that is already crowded with icons. A node can have a whopping 5 icons when it's in a group, has a connected signal, has a script, is visible, and now is enabled.

**Comment by passivestar** [comment]
Hey, I like it

I have some questions:

1. Would it make sense for this button to also toggle visibility in editor? If editor and runtime mirror each other you'll always know what's gonna happen when you press play and there will be less _"where did my stuff go? oh that's right I disabled it"_ moments. It's not immediately obvious what the value of seeing disabled stuff in the viewport is. I know you can make it a habit to click both, but it's strictly a UX question
2. Does this also disable editor `@tool`s? If I have a disabled tool node saved as a scene, will its `_enter_tree()` run when I instantiate it in editor?
3. Are you sure about the diamond shape? It's often associated with keyframing in software. What do you think of a rounded square instead?

**Comment by yoont4** [comment]
Would really like to see this go in or something similar. I've been wanting an easy way to test scenes with nodes on/off without having to completely destroy it and lose all data, or make any editor plugin hacks to work around it.

**Comment by BliznyukNM** [comment]
I was thinking about UX problem @AdriaandeJongh mentioned. One solution I can think of is having 3-state visibility button - visible, hidden, disabled (as disabled is hidden at the same time). Eye icon in hidden state can be filled half-way to illustrate that there is yet another click possible.

And to go back to visible state again you would have to cycle through all states.

**Comment by AdriaandeJongh** [comment]
Does the icon really need to be in the SceneTree for quick access? I can imagine it is initially enough to having it in the inspector together with the Node properties. If then enough people use the feature and want 'quick access' to it, then a followup PR could add the icon.

**Comment by grgp** [comment]
I think a check icon just like in Blender/Unity would work best? In Blender, it's also possible to filter the icons that you want to see on the right, but that's probably a separate PR.

image

image

For now I think I agree with AdriaandeJongh that for now it can be placed at the very top of the inspector, just like in Unity. I think this feature is direly needed, so the faster it can be merged in, the better.
image

**Comment by KoBeWi** [comment]
See also #56446

**Comment by AThousandShips** [comment]
How does this close https://github.com/godotengine/godot-proposals/issues/3433? I don't see any editor only or game only options for this.

**Comment by LauraWebdev** [comment]
Having this as a quick button in the SceneTree would be a game changer for my productivity. I personally enjoy the "three state button" idea shared to combine visibility and activity, although there are also valid reasons to not combine them (one being a changed workflow for those not interested in this feature)

**Comment by AThousandShips** [comment]
@LauraWebdev Please don't bump without contributing significant new information. Use the :+1: reaction button on the first post instead.

**Comment by torcado194** [comment]
@AdriaandeJongh yep, like mentioned in the post this could alternatively be hidden under the right-click menu to reduce the clutter. personally, i prefer it immediately accessible (for prototyping, etc.). I don't have any strong opinions one way or the other, this was simply what I ended up preferring.
IMO, I think this should be the default behavior of the eye icon, and disabling drawing should be hidden away. I consider making a node invisible but still active a more niche use case than toggling its existence entirely. But that's a much more radical change that I wouldn't expect to be considered :P 

@passivestar 
1. Part of the point of this feature is that it enabled **editor-only** nodes. I can use this to place visual markers and indicators in scenes and have them not exist in the built project. If you want something completely removed, you disable both icons, they work in tandem. Understandably there's a disconnect here between what you see in-editor and in-game, but that's the point. otherwise, like i said, this shouldn't be a separate toggle and should replace the eye icon behavior. Changing this would remove 50% of the usability for me.
2. Yes, it doesn't impact node's state in the editor, only when building. If you disable existence on a `@tool` node, it simply removes it from the built project. It still exists and runs in-editor.
3. I don't have any strong connection to the icon. It's simply what I ended up using. There's tons of discussion about the UI design in https://github.com/godotengine/godot-proposals/issues/7715, but no consensus was made. Personally, I think merging this behavior into the eye icon (so that it has 3+ states) is clever but unnecessarily confusing for people not understanding what it does. However, if this feature were hidden behind the right-click menu, I think it would be nice to have this existence toggle controlled by ctrl+clicking the eye icon (or another modifier key). 
Here's a demo:
demo

@AThousandShips By disabling existence and not visibility, a node remains visible and active in the editor but not in the built project. This is done automatically. However, this doesn't implement the inverse behavior (the node exists in the built project but not in the editor). So I don't believe this fully closes that proposal. I've edited the main post.

**Comment by passivestar** [comment]
@torcado194 What about renaming "Toggle Existence" to "Toggle Editor-Only" to make it unambiguous?

**Comment by torcado194** [comment]
@passivestar I think that's fine, yeah. It didn't register to me that that is what this is ultimately doing.

Alternatively, I've considered the idea of a 3-state existence toggle: "editor-only", "game-only", and "disabled". This would end up closing https://github.com/godotengine/godot-proposals/issues/3433, maybe as a separate PR. But I do think I want to implement this for myself now haha

**Comment by passivestar** [comment]
> if this feature were hidden behind the right-click menu, I think it would be nice to have this existence toggle controlled by ctrl+clicking the eye icon

What if:

- `Click` - Toggle visibility
- `Shift + Click` - Toggle editor-only
- `Ctrl + Click` - Toggle both (if they are in different states ctrl+clicking would first bring them to the same state based on the new visibility state)

This would make both workflows one-click

**Comment by AdriaandeJongh** [comment]
Toggling visibility and toggling whether a node is editor-only are two wholly different concepts to me. On top of that and unlike toggling visibility, making a node editor only has a lot of far reaching consequences such as non-editor-only nodes not being able to reference them. Such consequences make me vote strongly against putting it under / alongside the visibility toggle.

I like the feature / idea a lot, but my suggestion would be to put the option to make a node editor-only under the Node category, alongside the fields for process and metadata and such.

**Comment by torcado194** [comment]
@AdriaandeJongh I think those points are fair. I feel like these different solutions are circularly conflicting (having a button on the node tree allows for quick prototyping but adds clutter, removing the button but adding a shortcut to toggle it removes the clutter but semantically couples "editor-only" and "visibility" when they should be distinct, moving the button to the inspector decouples the properties but removes the quick toggle).

I prefer some way to quickly toggle the state because prototyping is my main use case for this feature, but if the button were moved to the inspector panel it wouldn't really matter to me, i'd just make a plugin to add a hotkey/button to switch it anyway. Whatever the consensus is, I'm happy to make changes :)

**Comment by passivestar** [comment]
> I prefer some way to quickly toggle the state because prototyping is my main use case for this feature

I think having this function in the context menu and exposed to editor's shortcuts to be able to bind some hotkey to it (can be empty by default) is the best compromise here. Then you don't need to make an addon

**Comment by AThousandShips** [comment]
Let's discuss the more general details of the feature on the proposal, and strictly the implementation details here

**Comment by Snowdrama** [comment]
> Personally, I think merging this behavior into the eye icon (so that it has 3+ states) is clever but unnecessarily confusing for people not understanding what it does. 

> I've considered the idea of a 3-state toggle: "editor-only", "game-only", and "disabled". 

@torcado194 For me the red text indicates that there's an "issue" with the node so I personally would avoid that just because for me it indicates that I need to check it to see if something is broken, like a missing resource reference or something.

I do think it needs some kind of visual that clearly shows it's somehow different than other nodes without looking like the node is somehow broken or has an error. 

If the icon has multiple states, I think it would make sense to change the icon itself as well as the hover tooltip. So for example changing the eye to a dark grey E that says "Editor-Only" on hover or something similar, while the eye icons continue to say "Toggle Visibility" on hover. 

> What if:
> 
> * `Click` - Toggle visibility
> * `Shift + Click` - Toggle editor-only
> * `Ctrl + Click` - Toggle both (if they are in different states ctrl+clicking would first bring them to the same state based on the new visibility state)

@passivestar Adding a shortcut to speed up toggling it would definitely be nice 100%, my only thing would be that the functionality would be hidden behind a shortcut and the information for it isn't surfaced anywhere in the editor itself which I think is a valuable even if there is a shortcut. I think that would be solved by @AdriaandeJongh's suggestion of putting it in the Inspector with the Process drop down, a tooltip could be put there saying something like "This can also be toggled by Ctrl+Clicking the Visibility toggle icon in the scene view" or something. 

> I like the feature / idea a lot, but my suggestion would be to put the option to make a node editor-only under the Node category, alongside the fields for process and metadata and such.

@AdriaandeJongh I personally like having it in the scene tree because being able to tell what is set to editor-only is useful to be able to at a glance see what won't be added to the build. Putting the toggle in the inspector view with process would be fine, but I would still have some kind of visual indicator to show that it is editor-only in the scene view.

**Comment by Kalto-Mate** [comment]
This is absolutely needed, but the whole icon debate is making another problem surface: There's already a LOT of icon clutter and it's a creeping problem, so one thing that does come to mind is to have the visibility toggle house node disabling functionality. 

Basically, you would cicle from visible to hidden to disabled, then back to visible. Maybe eye, closed eye and forbidden symbol. My reasoning is that I don't see how anyone would need to make a node invisible if they are going to also disable it. Double clicking MIGHT get annoying, so maybe other mouse buttons could cicle backwards if used on top of the button.

I know I'm glossing over the disabling scope nuance (editor only, game only) but I feel it might be justifiable for this to be buried just under the surface instead of a button on the tree. After all is an specific and deliberate technical choice and you really don't want to set stuff to editor only on accident and then be confused. To that extent a node set to be editor only could rock a "G" within a forbidden sign instead of the plain forbidden sign with nothing inside (which means disabled across the board)

image

**Comment by PLyczkowski** [comment]
I would go with a dropdown menu at the top of the inspector with Enabled/Disabled/Editor-only (RMB context menu is also fine but less discoverable), not adding a clickable icon, and signifying state in the scene tree view with graying out the name for disabled and coloring it for editor-only (I think yellow is taken, so maybe blue?)

**Comment by torcado194** [comment]
I have made changes following the suggestions made here and in the linked proposal.

- Removed the icon from the scene tree
- Added a button to the context menu of tree items to toggle editor-only (which also allows for an optional shortcut for the toggle)
- Marked editor-only nodes as *blue* (and descendants faded)
- Changed implementation to use `editor_only` in place of `disabled` internally
- Fixed formatting

I think it also makes sense to rename this PR to "Add Editor-Only Toggle", let me know if there's any reason I shouldn't do that.

Here is a demo of the changes:
(The shortcut is empty by default, I added one just for the demo)

demo2

**Comment by Kalto-Mate** [comment]
I fear using color to convey information, SPECIALLY blue is a big no-no for accesible design. I have no idea how much control over the text formatting on node's names is exposed, but would bold, italics, crossed-through or even color flipped presentation (see image) be more prudent?

623tge7289

On a side note, maybe it's just me, but back when I used Unity I did some stuff by having their equivalent of nodes disabled full stop and re-enabling them at runtime, so settling with just editor only disabling does narrow possibilities a bit.

**Comment by Calinou** [comment]
> I have no idea how much control over the text formatting on node's names is exposed, but wouldn't bold, italics, crossed-through or even color flipped presentation (see image) be more prudent?

To my knowledge, you can't use different font styles for individual tree items, but you can give them a dedicated background color. This is already done in the FileSystem dock for colored folders.

**Comment by KoBeWi** [comment]
> To my knowledge, you can't use different font styles for individual tree items

*Technically* you can use custom draw for that and draw the text manually.

**Comment by Kalto-Mate** [comment]
> I have made changes following the suggestions made here and in the linked proposal.

Just curious, is hijacking the eye button like I suggested really far out there and off the table?

**Comment by Macklehatton** [comment]
> Just curious, is hijacking the eye button like I suggested really far out there and off the table?

Look at #7715 and #3433 for context and more theoretical discussion.

**Comment by PLyczkowski** [comment]
Oh, blue is already used for editor-running script icons, did not notice that. I guess it either makes blue a better candidate for editor-only nodes from this patch, or if it is rejected due to accessibility then the editor-running script icons should be looked at as well (they could have a T letter on the icon for example).

image

**Comment by passivestar** [comment]
@PLyczkowski it's not blue it's the accent color:

image

**Comment by PLyczkowski** [comment]
> @PLyczkowski it's not blue it's the accent color

Oh. Not sure that's a good implementation then, no way to remember 'tool scripts look like this'... Icon with added gear or the letter T would be better imo

**Comment by adamscott** [comment]
Sorry for the delay.

I love the idea of a node existence toggle, but I don't really like the fact that it's closely binded to the editor.
In my mind, it should be possible to do this from code too. It would add complexity to the feature, though.

**Comment by Kalto-Mate** [comment]
> Sorry for the delay.
> 
> I love the idea of a node existence toggle, but I don't really like the fact that it's closely binded to the editor. In my mind, it should be possible to do this from code too. It would add complexity to the feature, though.

As I understand it, this proposal gets the job done by doing some hacky procedures to get nodes to actually cease to exist. The moment disabling becomes an official clean feature, like a boolean you can set, both code use and gui implementation will be easier.

**Comment by Snowdrama** [comment]
> As I understand it, this proposal gets the job done by doing some hacky procedures to get nodes to actually cease to exist. 

Based on what I've read it adds a new value to the base node class that when set, makes it so the routine to instantiate a packed scene doesn't actually instantiate the nodes marked as "editor-only" if the scene is being loaded in a build.

>  it should be possible to do this from code too

I think adding code features could be added later, simply because adding the "editor-only" flag at runtime would not have any effect since this affects the Packed Scene loading routine. So making it accessible via code would only add the ability for tool scripts to set the flag in the editor, and while that could be useful, I don't think it should be a blocker.

There's another proposal that is discussing something that is different, since the node is loaded but can be "enabled/disabled" I think there it makes way more sense since the goal is to make it able to be toggled at runtime:
https://github.com/godotengine/godot-proposals/issues/7715

The only other additional code driven enhancement might also be an additional flag in GDScript like `@editor_only` that is similar to how `@tool` works, and would just force on the "editor_only" flag on assignment, but this wouldn't get rid of the ability to set it in the editor, and I don't think this should inherently be a blocker either.

**Comment by conde2** [comment]
> I think a check icon just like in Blender would work best? In Blender, it's also possible to filter the icons that you want to see on the right, but that's probably a separate PR.
> 
> image image
> For now I agree with AdriaandeJongh that for now it can be placed at the very top of the inspector, just like in Unity. image
> 
> Also worth noting that if you disable a gameobject in Unity, the text on the scene tree will be greyed out. image
> 
> I think this feature is direly needed, so the faster it can be merged in, the better.

I also prefer this solution over the others persented here, having it in the inspector with a simple checkbox is enought, clean and easy to access.

**Comment by geekley** [comment]
I think whatever implementation is used for an "editor-only" feature, it should be at least forward-thinking about possible future extension to a more general "availability" feature, of which "editor-only" is just a subset. Plus "editor-only" is not specific enough, as you may want different definitions for it.

So ideally we should be able to set any availability, to define when nodes are included or excluded completely from builds. Luckily there's already a feature tags concept, which should be perfect for this, e.g.:
- this node only exists in the editor when not running: `metadata/availability = "editor_hint"`
- this node only exists at runtime (including when running from the editor): `metadata/availability = "!editor_hint"`
- this node only exists at runtime when running from the editor: `metadata/availability = "editor_runtime"`
- this node only exists on the editor (whether or not it's running): `metadata/availability = "editor"`
- this node only exists on exported games: `metadata/availability = "template"`
- this node only exists on mobile platforms: `metadata/availability = "mobile"`
- this node only exists on (exported?) desktop/laptop platforms: `metadata/availability = "pc"`
- this node only exists on web platform: `metadata/availability = "web"`
- this node only exists on my custom feature tag (e.g. per-store builds): `metadata/availability = "google_play"`

Basically, it should set a feature tag representing the builds where the node can be included, or a preceding `!` to include it everywhere except on builds with that tag. I'm talking about how this is serialized -- editor UI could be just a simple toggle for `"editor_hint"` VS `""` (everywhere) for now, and implementation expanded later.

This is a flexible implementation, since you could also in the future allow maybe even expressions with `|` and `&` in the string, e.g. `"pc & !editor"`, and its format would be compatible.

Also, this concept could be later expanded to resources as well.

**Comment by torcado194** [comment]
> I think whatever implementation is used for an "editor-only" feature, it should be at least forward-thinking about possible future extension to a more general "availability" feature...

I think there's too much confusion and contradiction around what people expect this to solve, and I genuinely think is the reason why any PR trying to implement this or a similar feature will fail to be merged. It's difficult to be clear about the intention of this PR. It's very possible there's a more applicable name for this feature, it's already been renamed a couple times.
This PR is not trying to create a general solution for including or excluding nodes/resources/objects from exported builds. There are several existing features that touch that (e.g. feature tags, as you noted) and many many proposals that relate to that idea in some way.
The best comparison I have to explain this PR is commenting out code in a script. A programmer can quickly comment out lines of code to test different behaviors. This is the same concept but with scene nodes in the editor.
- The problem is that quickly a/b testing scene structures is tedious and error prone.
- The proposed solution (and the solution this PR implements) is to add a toggle that simply excludes nodes in the editor from the built project.  

This is purely an editor quality of life feature. This does not need to support targeting platforms, etc. and I think trying to expand the scope to solve potential future problems will make this PR even *less* likely to be merged.

I think the idea of including/excluding nodes based on platform or export context is useful, but unrelated to this PR.

File: scene/resources/packed_scene.cpp [hunk 1]
```diff
@@ -336,6 +350,15 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 
 					ERR_FAIL_INDEX_V(nprops[j].name, sname_count, nullptr);
 
+					if (!Engine::get_singleton()->is_editor_hint()) {
+						if (snames[nprops[j].name] == "metadata/_editor_only_" && props[nprops[j].value]) {
+							editor_only = true;
+							node = Object::cast_to<Node>(memnew(Node));
```
Comment (AThousandShips) [inline scene/resources/packed_scene.cpp]:
```suggestion
							node = memnew(Node);
```

File: scene/resources/packed_scene.cpp [hunk 2]
```diff
@@ -191,7 +191,19 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
...
+		if (!Engine::get_singleton()->is_editor_hint()) {
+			editor_only = n.editor_only > 0;
+			if (parent) {
+				if (parent->get_scene_editor_only()) {
+					editor_only = true;
+				}
+			}
+		}
+		if (editor_only) {
+			node = Object::cast_to<Node>(memnew(Node));
```
Comment (AThousandShips) [inline scene/resources/packed_scene.cpp]:
```suggestion
			node = memnew(Node);
```

**Comment by PLyczkowski** [comment]
Unity's implementation is a bit different btw - Unity’s GameObject.SetActive(bool) also toggles a GameObject’s active/inactive state at runtime or play mode, stopping execution and hiding it - they stop receiving Update(), physics, rendering, and event callbacks.

This has an additional benefit over the implementation proposed here that one can toggle active state of a game object and it's children at runtime for testing purposes. Not sure if it's viable to implement it in Godot like that due to it's node based design, as opposed to the Unity's game objects + components, but one can simulate the behavior by calling:

set_process(false)
set_physics_process(false)
visible = false
set_block_signals(true)

But there’s no single “active/inactive” flag that halts all of them simultaneously.

In summary: This feature would become even more useful if it allowed for runtime toggling.

**Comment by RobProductions** [comment]
> But there’s no single “active/inactive” flag that halts all of them simultaneously.
> In summary: This feature would become even more useful if it allowed for runtime toggling.

@PLyczkowski This is exactly what I tried to accomplish with https://github.com/godotengine/godot/pull/94144 (and could even work in tandem with this PR) but it seems some maintainers were hesitant about the idea. I never really understood why exactly and the best I got what was "it's just not possible to do this in Godot" or something along the lines of "Godot doesn't work like that" which hasn't made sense because clearly it CAN work like that as I've tried to demonstrate with the PR. Questions about how it works "philosophically" with Godot's workflow are just as confusing to me as user expectations can change and be changed especially when aided by proper communication about new features.

A replacement idea called Node Stashing came up here: https://github.com/godotengine/godot-proposals/issues/10212 but it was created without any input from me and I believe it's much harder to implement and harder to use, being that user code will have to check the "stashed" list, it won't be compatible with animations, and won't work on scene roots. Truthfully I haven't tried much to push for answers about why Node Active isn't viable because I was a bit burned by how quickly it was dismissed and the fact that no one reached out to me when they tried to theorize an improvement. But the few times I have tried to ask about it were met with silence. If we want to open up discussions about this feature again, I'd be all for it if it would finally provide some answers about the old PR and the ways it can be improved without bypassing the core idea that's supposed to make it simple to understand in the first place.

**Review by ghost — APPROVED** [review]

**Comment by LauraWebdev** [comment]
> Unity's implementation is a bit different btw - Unity’s GameObject.SetActive(bool) also toggles a GameObject’s active/inactive state at runtime or play mode, stopping execution and hiding it - they stop receiving Update(), physics, rendering, and event callbacks.
> 
> This has an additional benefit over the implementation proposed here that one can toggle active state of a game object and it's children at runtime for testing purposes. Not sure if it's viable to implement it in Godot like that due to it's node based design, as opposed to the Unity's game objects + components, but one can simulate the behavior by calling:
> 
> set_process(false) set_physics_process(false) visible = false set_block_signals(true)
> 
> But there’s no single “active/inactive” flag that halts all of them simultaneously.
> 
> In summary: This feature would become even more useful if it allowed for runtime toggling.

In a way, the "Godot" way would probably look like a "process_mode" of "editor_only"? I'm following this issue for a while now and just thought of this.

File: editor/scene_tree_dock.cpp [hunk 3]
```diff
@@ -3933,50 +3828,42 @@ void SceneTreeDock::_tree_rmb(const Vector2 &p_menu_pos) {
...
 			}
 		}
-		END_SECTION()
+
+		if (add_separator && profile_allow_editing) {
+			menu->add_separator();
+		}
 	}
 
+	menu->add_icon_shortcut(get_theme_icon(SNAME("GuiExistenceDisabled"), SNAME("EditorIcons")), ED_GET_SHORTCUT("scene_tree/editor_only"), TOOL_EDITOR_ONLY);
```
Comment (AThousandShips) [inline editor/scene_tree_dock.cpp]:
```suggestion
	menu->add_icon_shortcut(get_editor_theme_icon(SNAME("GuiExistenceDisabled")), ED_GET_SHORTCUT("scene_tree/editor_only"), TOOL_EDITOR_ONLY);
```

File: scene/resources/packed_scene.cpp
Comment (AThousandShips) [inline scene/resources/packed_scene.cpp]:
```suggestion
			node = memnew(Node);
```
Redundant

Comment (AThousandShips) [inline scene/resources/packed_scene.cpp]:
```suggestion
							node = memnew(Node);
```
